[Edit] Update operators.md (#1210)
Codecademy/docs: content/javascript/concepts/operators/operators.md

### Multi-Step Ternary Operator Examples

Multiple steps can be performed by a ternary operator, based on certain conditions (similar to an `if...else` statement).

A key point to consider is where the parentheses (`()`) are placed.

The following example shows how a ternary operator can conditionally increment numeric values similar to how it's done in an `if...else` statement:

```js
let i = true;
let a = 0;
let b = 0;

if (i == true) {
  a++;
  b++;
} else {
  a = 0;
  b = 0;
}

console.log('a = ', a, 'b = ', b);

// The ternary equivalent
i == true ? (a++, b++) : ((a = 0), (b = 0));

console.log('a = ', a, 'b = ', b);

let c = 0;
i == true ? (a++, b++, c++) : ((a = 0), (b = 0), (c = 0));

console.log('a =', a, 'b =', b, 'c =', c);
```

This will output the following:

```shell
a = 1 b = 1
a = 2 b = 2
a = 3 b = 3 c = 1
```

The following ternary operator will throw a `SyntaxError` due to how the parentheses are placed:

```js
let i = true;

// This implementation will fail
i == true ? (a++), (b++) : (a = 0), (b = 0);
console.log('a = ', a, 'b = ', b);
```

This will output:

```shell

> // Output:
> // i == true ? (a++), (b++) : (a = 0), (b = 0);
> //                  ^
> // SyntaxError: Unexpected token ','

Reviewer: ```suggestion
Output:
i == true ? (a++), (b++) : (a = 0), (b = 0);
                           ^
SyntaxError: Unexpected token ','
```

> // Output:
> // i == true ? (a++), (b++) : (a = 0), (b = 0);
> //                  ^
> // SyntaxError: Unexpected token ','

Reviewer: ```suggestion
Output:
i == true ? (a++), (b++) : (a = 0), (b = 0);
                 ^
SyntaxError: Unexpected token ','
```

```

> Although ternary operator and `if...else` condition performance speeds are [roughly the same](https://stackoverflow.com/questions/2586842/is-ternary-operator-if-else-or-logical-or-faster-in-javascript#:~:text=test%20was%20fair-,Here%27s%20my%20jsperf,-result%3A), using multi-step ternary operators can help consolidate several lines of code, thus reducing file size and download time in the browser.

Reviewer: ```suggestion
> **Note:** The ternary operator and `if...else` condition performance speeds are roughly the same. Although the ternary operator can help consolidate several lines of code, utilizing it is up to preference.
```

Reviewer: @Dusch4593 How about we change this to a note?

Reviewer 2: I'd be open to a note 😄

This operator combines the functionalities of the assignment, comparison, and logical operators.

> The ternary operator can also perform multiple steps based on a certain condition, similar to an `if...else` condition statement.

Reviewer: @Dusch4593 Can we instead take the text here and move it to the next heading which goes over multi-step ternary operators?

> The key point to focus on is the syntax. Pay special attention to where the `()`s are placed. A developer may make the mistake of using too many `()`s or placing the `()`s incorrectly.
> 
> ### Multi-Step Ternary Operator Examples
> 
> For example, the following `if...else` statement...

Reviewer 2: ```suggestion
A key point to consider is where the parentheses (`()`) are placed.

The following example shows how a ternary operator can conditionally increment numeric values similar to how it's done in an `if...else` statement:
```

```js
let i = true;
let a = 0;
let b = 0;

if (i == true) {
    a++;
    b++;
} else {
    a = 0;
    b = 0;

> }

Reviewer 2: ```suggestion
}

console.log('a =', a, 'b =', b)

// This does the same thing
i == true ? (a++, b++) : (a = 0, b = 0);

console.log('a =', a, 'b =', b)

let c = 0
i == true ? (a++, b++, c++) : (a = 0, b = 0, c = 0);

console.log('a =', a, 'b =', b, 'c =', c)
```

I think we should consolidate some of these \```js blocks into a single block that walks through how (a) a ternary operator behaves in a similar way to an `if...else` statement, and (b) multiple steps can be introduced to ternaries.

```

> ...can be translated into the following two-step ternary operator!
> 
> ```js
> let i = true;
> let a = 0;
> let b = 0;
> 
> i == true ? (a++, b++) : (a = 0, b = 0);
> console.log('a = ', a, 'b = ', b);
> ```
> 
> The ternary operator can also perform as many steps as you need.
> 
> ```js
> let i = true;
> let a = 0;
> let b = 0;
> let c = 0;
> // ...
> 
> i == true ? (a++, b++, c++) : (a = 0, b = 0, c = 0);
> console.log('a = ', a, 'b = ', b, 'c = ', c);
> ```

Reviewer 2: ```suggestion
This will output the following:
```

And then we should add a \```shell output snippet that looks like this:

```
a = 1 b = 1
a = 2 b = 2
a = 3 b = 3 c = 1
```

> The following ternary operator will result in a SyntaxError. Do you see why?

Reviewer 2: ```suggestion
The following ternary operator will throw  a `SyntaxError` due to how the parentheses are placed:
```

```js
let i = true;

// This implementation will fail
i == true ? (a++), (b++) : (a = 0), (b = 0);
console.log('a = ', a, 'b = ', b);

> 
> // Output:
> // i == true ? (a++), (b++) : (a = 0), (b = 0);
> //                  ^
> // SyntaxError: Unexpected token ','

Reviewer 2: ```suggestion
```

Reviewer 2: This should be separated into its own code snippet.

```

> > Hint: check how the `()`s were implemented.
> 
> 

Reviewer 2: ```suggestion

```

Reviewer 2: Here, I would suggest adding a \```shell snippet with the thrown error message:

```
 i == true ? (a++), (b++) : (a = 0), (b = 0);
                            ^
SyntaxError: Unexpected token ','
```

console.log(isDuck);
```

The ternary operator can also perform multiple steps based on a certain condition, similar to the IF/ELSE condition statement.

> For example, the following IF/ELSE statement...

Reviewer 3: ```suggestion
For example, the following `if...else` statement...
```

Author: Thank you @SSwiniarski for letting me know about these changes! I'm uploading them now.

```js
let i = true;
let a = 0;
let b = 0;

if (i == true) {
    a++;
    b++;
} else {
    a = 0;
    b = 0;
}
```

...can be translated into the following two-step ternary operator!

```js
let i = true;
let a = 0;
let b = 0;

i == true ? (a++, b++) : (a = 0, b = 0);
console.log('a = ', a, 'b = ', b);
```

> The ternary operator can also perform as many steps as you prefer.

Reviewer 3: ```suggestion
The ternary operator can also perform as many steps as needed.
```

Author: Thank you @SSwiniarski for letting me know about these changes! I'm uploading them now.

```js
let i = true;
let a = 0;
let b = 0;
let c = 0;
//...

i == true ? (a++, b++, c++) : (a = 0, b = 0, c = 0);
console.log('a = ', a, 'b = ', b, 'c = ', c);
```

The key point to focus on is the syntax. Pay special attention to where the `()`s are placed. A developer may make the mistake of using too many `()`s or placing the `()`s incorrectly, as shown below.

The following ternary operator will result in a SyntaxError:

```js
let i = true;

// This implementation will fail
i == true ? (a++), (b++) : (a = 0), (b = 0);
console.log('a = ', a, 'b = ', b);

// Output:
// i == true ? (a++), (b++) : (a = 0), (b = 0);
//                  ^
// SyntaxError: Unexpected token ','
```

> Although ternary operator and IF/ELSE performances speeds are [roughly the same](https://stackoverflow.com/questions/2586842/is-ternary-operator-if-else-or-logical-or-faster-in-javascript#:~:text=test%20was%20fair-,Here%27s%20my%20jsperf,-result%3A), using multi-step ternary operators can help consolidate several lines of code, thus reducing file size and download time in the browser.

Reviewer 3: ```suggestion
Although ternary operator and `if...else` performances speeds are [roughly the same](https://stackoverflow.com/questions/2586842/is-ternary-operator-if-else-or-logical-or-faster-in-javascript#:~:text=test%20was%20fair-,Here%27s%20my%20jsperf,-result%3A), using multi-step ternary operators can help consolidate several lines of code, thus reducing file size and download time in the browser.
```

Author: Thank you @SSwiniarski for letting me know about these changes! I'm uploading them now.
